var/spack/repos/builtin/packages/bml/package.py

```diff
@@ -38,12 +38,23 @@ class Bml(CMakePackage):
     version('develop', git='https://github.com/lanl/bml', branch='master')
 
     variant('shared', default=True, description='Build shared libs')
+    variant('mpi', default=False, description='Build with MPI Support')
+
+    conflicts('+mpi', when='@:1.2.2')
 
     depends_on("blas")
     depends_on("lapack")
+    depends_on('mpi', when='+mpi')
 
     def cmake_args(self):
-        return [
+        args = [
             '-DBUILD_SHARED_LIBS={0}'.format(
                 'ON' if '+shared' in self.spec else 'OFF')
         ]
+        spec = self.spec
+        if '+mpi' in spec:
+            args.append('-DBML_MPI=True')
+            args.append('-DCMAKE_C_COMPILER=%s' % spec['mpi'].mpicc)
+            args.append('-DCMAKE_CXX_COMPILER=%s' % spec['mpi'].mpicxx)
+            args.append('-DCMAKE_Fortran_COMPILER=%s' % spec['mpi'].mpifc)
```

Comment:
maybe add a `args.append('-DBML_MPI=False')` in an `else:` block in case bml's default changes as some point.